src/cpp/session/modules/clang/DefinitionIndex.cpp

```diff
@@ -83,7 +83,8 @@ CXChildVisitResult cursorVisitor(CXCursor cxCursor,
 
    // ensure it's a definition with linkage or a typedef
    if ((!cursor.isDefinition() || !cursor.hasLinkage()) &&
-       cursorKind != CXCursor_TypedefDecl)
+       cursorKind != CXCursor_TypedefDecl &&
+       cursorKind != CXCursor_MacroDefinition)
```

Comment:
One thing this should maybe do is attempt to remove `define` that are for header guards. Remove macros when the name ends with `_H` or `_HPP` ?

Reply:
That sounds like a reasonable heuristic to me

I think most of us use that pattern:
cli - https://github.com/r-lib/cli/blob/c39f7957636286562a78d9e6e63abed038298a1a/src/cleancall.h#L2
dplyr - https://github.com/tidyverse/dplyr/blob/138f9fe5bfa8004fa4e7fb8bbd11a15bfa70fcbe/src/dplyr.h#L2
vctrs - https://github.com/r-lib/vctrs/blob/0a219ba242ee368df63bc3876f06f9a6220e85c0/src/altrep-rle.h#L2

I am sure there are some that don't follow this, but I think that is ok